torch/nested/_internal/ops.py

```diff
@@ -536,6 +569,52 @@ def is_same_size_default(func, *args, **kwargs):
     return args[0]._size == args[1]._size
 
 
+@register_jagged_func(
+    torch.ops.aten.sum.dim_IntList, "self: jt, dim: any?, keepdim: any?, dtype: any?"
+)
+def sum_dim_IntList(func, *args, **kwargs):
+    from torch._prims_common import canonicalize_dims
+
+    # sum_dim_IntList can produce a NT or a T depending on whether the ragged dim
+    # is summed over
```

Comment:
I think returning plain T has more known benefits, and it's not clear if returning NT today would give us more flexibility wrt bc either. If we return NT today and make users to grab values explicitly, that would no longer work if we choose to return T later.

Pros:
- avoid having additional state on NT tracking whether we are a dense NT or not
- less additional logic within autograd to handle the conversion from dense NT to dense
- users would not have to worry about having to manually converting the dense NT back to dense to avoid subclass overhead

Cons:
- some ops on NT unexpectedly(?) return non-NTs. I'm not actually sure how problematic it is. On the NT front, as long as NTs can freely promote, doing any NT related ops should not be problematic on this dense T.

Reply:
cc @albanD

Reply:
From offline discussion @albanD thinks this seems fine as long as there are no silent correctness issues.

Reply:
Cool I'm good with it then :)